Merge create callback and create subscrition

dfarr opened the pull request:
These will be separate api subsystem requests but the same coroutine.

I intended to change the routes to:
```
/promises/{id}/callback
/promises/{id}/subscribe
```

However we use a catch-all for the id parameter, which means the id may include a "/" and additional parts cannot be matched following this parameter. I have left the http routes as-is.

dfarr commented on internal/app/coroutines/createCallback.go:
```diff
@@ -1,22 +1,25 @@
+	util.Assert(r.CreateCallback.Mesg.Type == "resume" || r.CreateCallback.PromiseId == r.CreateCallback.Mesg.Root, "if notify, root promise id must equal leaf promise id")
+	// util.Assert(r.CreateCallback.Mesg.Type == "notify" ||(r.CreateCallback.PromiseId == r.CreateCallback.Mesg.Leaf && r.CreateCallback.PromiseId != r.CreateCallback.Mesg.Root), "if resume, root promise id must not equal leaf promise id")
+
+	// TODO: should this be a validation in the api layer?
```
@avillega how do you feel about moving this to the api layer?

avillega replied:
I think I prefer it in the coroutine. It is kind of a business logic validation, and it applies to all api subsystems. we would have to do the validation in the http subsystem and grpc subsystem and any future ones. I prefer when all the validations are close together so it is easy to track where stuff fails.

dfarr replied:
I agree that it would be preferable to centralize the validation. The tricky thing is I would also prefer to "fail fast". 

Hmmm okay what about building out a validation mechanism in the kernel api implemented [here](https://github.com/resonatehq/resonate/blob/main/internal/api/api.go#L136)? I imagine we could have hooks based on the request kind, and the best part is, this is the interface that [dst uses](https://github.com/resonatehq/resonate/blob/main/test/dst/dst.go#L152) too!

avillega reviewed (approved):
Don't forget to update the openapi.yml file. I am unsure if this could break the SDK given that the createcallback request does not accept an id anymore.

dfarr commented:
I have updated the open api spec and added api validators. I really like the api validators and suspect there are other things we could use them for, I figure we can do this in a step by step approach though.

dfarr commented on test/dst/dst.go:
```diff
@@ -658,6 +655,13 @@ func (d *DST) Step(model *Model, reqTime int64, resTime int64, req *t_api.Reques
+				// leaf promise ids by chance
+				return model, nil
+			}
+			fallthrough
```
I am both so happy and so horrified that I got to use a `fallthrough`
